File: crates/atlas-cli/tests/fixtures/go/class.go
package main

type Greeter interface {
	Greet() string
}

type Person struct {
	Name string
}

func (p *Person) Greet() string {
	return "Hello, " + p.Name
}

func main() {
	var g Greeter = &Person{Name: "World"}
	_ = g.Greet()
}

File: crates/atlas-cli/tests/fixtures/rust/class.rs
trait Greeter {
    fn greet(&self) -> String;
}

struct Person {
    name: String,
}

impl Greeter for Person {
    fn greet(&self) -> String {
        format!("Hello, {}", self.name)
    }
}

fn main() {
    let p = Person {
        name: "World".to_string(),
    };
    let _ = p.greet();
}
